src/common/map/MapDirective.js

```diff
@@ -53,15 +53,44 @@
 
             scope.$watch('layers', function(layers) {
               if (layers && map) {
-                var layerLength = map.getLayers().getLength();
-                for (var i = 1; i < layerLength; i++) {
-                  map.removeLayer(map.getLayers().getArray()[1]);
+                deleteNoCartLayers();
```

Comment:
First I delete the layers that were added in the map preview making hover through  `deleteNoCartLayers` function. 
In `addAllLayers` I delete only a repeated layer  with layerId property.